sched/scheduler/stateful_scheduler.go

```diff
@@ -838,7 +838,10 @@ func (s *statefulScheduler) scheduleTasks() {
 				nodeId := nodeSt.node.Id()
 				nodeStInstance, ok := s.clusterState.getNodeState(nodeId)
 				nodeAbsent := !ok
-				nodeReAdded := &nodeStInstance.readyCh != &nodeSt.readyCh
+				nodeReAdded := false
+				if !nodeAbsent {
+					nodeReAdded = &nodeStInstance.readyCh != &nodeSt.readyCh
```

Comment:
`nodeReAdded = (&nodeStInstance.readyCh != &nodeSt.readyCh)` would make this _slightly_ easier to read imo